src/components/BulkDeleteButton.tsx

```diff
@@ -0,0 +1,89 @@
+import { FC, Fragment } from "react";
+import { pluralize } from "util/instanceBulkActions";
+import {
+  ConfirmationButton,
+  ConfirmationButtonProps,
+  Icon,
+} from "@canonical/react-components";
+
+interface Props {
+  entities: unknown[];
+  deletableEntities: unknown[];
+  entityType: string;
+  onDelete: () => void;
+  disabledReason?: string;
+  extraActions?: string[];
+  confirmationButtonProps?: Partial<ConfirmationButtonProps>;
+  buttonLabel?: React.ReactNode;
+}
+
+const BulkDeleteButton: FC<Props> = ({
+  entities,
+  deletableEntities,
+  disabledReason,
+  entityType,
+  extraActions,
+  confirmationButtonProps,
+  onDelete,
+  buttonLabel,
+}) => {
+  const totalCount = entities.length;
+  const deleteCount = deletableEntities.length;
+
+  const actions =
+    extraActions?.map((action) => {
+      return (
+        <Fragment key={action}>
+          - {action}
+          <br />
+        </Fragment>
+      );
+    }) || [];
+
+  return (
+    <ConfirmationButton
+      className="u-no-margin--bottom has-icon"
+      appearance="base"
+      onHoverText={
+        disabledReason ?? `Delete ${pluralize(entityType, entities.length)}`
+      }
+      disabled={deleteCount === 0 || !!disabledReason}
+      shiftClickEnabled
+      showShiftClickHint
+      {...confirmationButtonProps}
+      confirmationModalProps={{
+        title: "Confirm delete",
+        children: (
+          <p>
+            {actions.length > 0 && (
+              <>
+                <b>{totalCount}</b> {pluralize(entityType, entities.length)}{" "}
+                selected:
+                <br />
+                <br />
+                {actions}
+                <br />
+              </>
+            )}
+            This will permanently delete <b>{deleteCount}</b>{" "}
+            {pluralize(entityType, deleteCount)}.{"\n"}This action cannot be
+            undone, and can result in data loss.
+          </p>
+        ),
+        confirmButtonLabel: "Delete",
+        onConfirm: onDelete,
+      }}
+    >
+      {buttonLabel ? (
+        buttonLabel
+      ) : (
+        <>
+          <Icon name="delete" />
+          <span>Delete</span>
+        </>
+      )}
```

Comment:
I think 
```suggestion
          <Icon name="delete" />
          <span>{buttonLabel}</span>
```
I think this should always render the icon and have the `buttonLabel` default to "Delete", but allow override for i.e. writing "Delete snapshot" or "Delete image". I'd suggest removing the loading states in the two snapshot cases to make the button label a string.